### DISCUSSION TITLE
freeKISS - Forecasting benchmark

### DISCUSSION THREAD

**Discussion by oo777** [alan-turing-institute/sktime#3056] [Ideas]
I'd like to provide a free forecast benchmark website utilizing sktime forecasters and some of our own proofed well recipes. The idea is to offer a keep it simple forecast result as a baseline/reference without login and personalisation/user analysis etc. Of course there will be some ressource limitations for cost reasons. May I ask for feedback:

a) You guys would use that or you think others would?
b) Can anyone provide a specification about TS file attributes (@)? I would like to implement that format too.

**Comment by fkiraly** [comment]
(a) @oo777, I think that would be a great idea that many people would find useful!
We have, in fact, been exploring benchmarking as part of one of our workstreams during the summer programme.
(regular meetings on Friday, see slack schedule, if you are interested to join and chat)

There are two separate concerns, (a1) python interface and (a2) a service or webpage that might use (a1) under the hood.
For (a1) we have explored a few options and are also working with @DBCerigo from `kotsu`.

For (a2), I presume the model would be something like `openML` where you can do that already for classification and other `sklearn` native learning tasks. From a design perspective and from a collaboration perspective, `openML` might be a good model.

(b) I have also asked for that earlier, this issue https://github.com/alan-turing-institute/sktime/issues/2043
since I felt we need to provide a precise specification if we want the file format to be used and interfaced by others for benchmarking.

I think, originally @TonyBagnall cooked it up based on the time series classification format he uses in his research group, but has never provided a full specification - at least he has never answered to the issue #2043.

Worst case, we reverse engineer a plausible specification and then declare that to be the "ground truth" - what do you think, @ltsaprounis, @oo777?

**Reply by oo777** [reply]
Thanks for your feedback!
Well with python in production I have little experience so far but I heard others do so. I started with R which works quite well in production batch mode. Since I discovered more and more aspects about ML and forecasting I will switch to python. Also sktime proofs me right I would say. Python would do most of the forecasting behind something serving all the needs of service architecture today. Thanks for mentioning OpenML which I totally forgot. I am not sure about utilizing one more framework but I will evaluate it. Finally let's say it's a working title and also my wife shared her concerns.

**Reply by oo777** [reply]
With the file specification I was thinking TS is the standard in the academic world but apperently not. So old school 90's csv files only than.

**Comment by fkiraly** [comment]
PS: kind recommendation re branding, if we do not integrate with an existing solution with an existing brand, like `openML`, I would strongly recommend to avoid acronyms with sexual connotation, such as your suggestion FreeKISS, or NIPS (the original name of the NeurIPS conference). Has a bit of a "schoolboy humour" touch and is reminiscent of outdated 1990s "boys only computer club" mentality.
